Sources/Purchasing/StoreKitAbstractions/SK2StoreTransaction.swift

```diff
@@ -25,6 +25,19 @@ internal struct SK2StoreTransaction: StoreTransactionType {
         self.quantity = sk2Transaction.purchasedQuantity
         self.jwsRepresentation = jwsRepresentation
 
+        if #available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *) {
+            self.environment = environment ?? StoreEnvironment(environment: sk2Transaction.environment)
+
+        } else {
+            #if VISION_OS
+            self.environment = environment ?? StoreEnvironment(environment: sk2Transaction.environment)
+            #else
+            self.environment = environment ?? StoreEnvironment(
+                environment: sk2Transaction.environmentStringRepresentation
+            )
+            #endif
+        }
```

Comment:
What do you think about extracting this to a new `StoreEnvironment` constructor that takes a `SK2Transaction`?

Then this becomes one line:
```swift
self.environment = environment ?? .init(sk2Transaction)
```